src/library/browse/browsetablemodel.cpp

```diff
@@ -130,9 +130,15 @@ QString BrowseTableModel::getTrackLocation(const QModelIndex& index) const {
 }
 
 TrackId BrowseTableModel::getTrackId(const QModelIndex& index) const {
-    Q_UNUSED(index);
-    // We can't implement this as it stands.
-    return TrackId();
+    TrackPointer pTrack = getTrack(index);
+    if (pTrack) {
+        return pTrack->getId();
+    } else {
+        qWarning()
```

Comment:
I think the track should be added to the library in this case. We do this for Banshee for example here: 
https://github.com/mixxxdj/mixxx/blob/master/src/library/banshee/bansheeplaylistmodel.cpp#L289

Reply:
getTrack() either gets the track from the library by location or adds it. The warning only appears if neither of both is possible.

Reply:
Ups, thank you for clarifying.